homeassistant/components/hassio.py

```diff
@@ -0,0 +1,250 @@
+    vol.Required(ATTR_ADDON): cv.slug,
+})
+
+SCHEMA_SERVICE_ADDONS_VERSION = SCHEMA_SERVICE_ADDONS.extend({
+    vol.Optional(ATTR_VERSION): cv.string,
+})
+
+
+SERVICE_MAP = {
+    SERVICE_HOST_SHUTDOWN: None,
+    SERVICE_HOST_REBOOT: None,
+    SERVICE_HOST_UPDATE: SCHEMA_SERVICE_UPDATE,
+    SERVICE_SUPERVISOR_UPDATE: SCHEMA_SERVICE_UPDATE,
+    SERVICE_HOMEASSISTANT_UPDATE: SCHEMA_SERVICE_UPDATE,
+    SERVICE_ADDON_INSTALL: SCHEMA_SERVICE_ADDONS_VERSION,
+    SERVICE_ADDON_UNINSTALL: SCHEMA_SERVICE_ADDONS,
+    SERVICE_ADDON_START: SCHEMA_SERVICE_ADDONS,
+    SERVICE_ADDON_STOP: SCHEMA_SERVICE_ADDONS,
+    SERVICE_ADDON_UPDATE: SCHEMA_SERVICE_ADDONS_VERSION,
+}
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup the hassio component."""
+    websession = async_get_clientsession(hass)
+    hassio = HassIO(hass.loop, websession)
+
+    if not hassio.connected:
+        _LOGGER.error("No HassIO supervisor detect!")
+        return False
+
+    # register api views
+    for base in ('host', 'supervisor', 'network', 'homeassistant'):
+        hass.http.register_view(HassIOBaseView(hassio, base))
+    hass.http.register_view(HassIOAddonsView(hassio))
+
+    @asyncio.coroutine
+    def async_service_handler(service):
+        """Handle HassIO service calls."""
+        addon = service.data.get(ATTR_ADDON)
+        if ATTR_VERSION in service.data:
+            version = {ATTR_VERSION: service.data[ATTR_VERSION]}
+        else:
+            version = None
+
+        # map to api call
+        if service.service == SERVICE_HOST_UPDATE:
+            yield from hassio.send_command(
+                "/host/update", payload=version)
+        elif service.service == SERVICE_HOST_REBOOT:
+            yield from hassio.send_command("/host/reboot")
+        elif service.service == SERVICE_HOST_SHUTDOWN:
+            yield from hassio.send_command("/host/shutdown")
+        elif service.service == SERVICE_SUPERVISOR_UPDATE:
+            yield from hassio.send_command(
+                "/supervisor/update", payload=version)
+        elif service.service == SERVICE_HOMEASSISTANT_UPDATE:
+            yield from hassio.send_command(
+                "/homeassistant/update", payload=version)
+        elif service.service == SERVICE_ADDON_INSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/install".format(addon), payload=version)
+        elif service.service == SERVICE_ADDON_UNINSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/uninstall".format(addon))
+        elif service.service == SERVICE_ADDON_START:
+            yield from hassio.send_command("/addons/{}/start".format(addon))
+        elif service.service == SERVICE_ADDON_STOP:
+            yield from hassio.send_command("/addons/{}/stop".format(addon))
+        elif service.service == SERVICE_ADDON_UPDATE:
+            yield from hassio.send_command(
+                "/addons/{}/update".format(addon), payload=version)
```

Comment:
I think we can do that later. for the moment it help for more readable of code and maby we change some things in next version. That is also the reasin of v1 on name :)